src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -80,7 +80,7 @@ public static bool ShouldSkipDeclarationAnalysis(ISymbol symbol)
         {
             // Skip syntax actions for implicitly declared symbols, except for implicitly declared global namespace symbols.
             return symbol.IsImplicitlyDeclared &&
-                !((symbol.Kind == SymbolKind.Namespace && ((INamespaceSymbol)symbol).IsGlobalNamespace));
+                !((symbol is INamespaceSymbol { IsGlobalNamespace: true, Kind: SymbolKind.Namespace }));
```

Comment:
>Kind: SymbolKind.Namespace [](start = 73, length = 26)

Semantically redundant test (the refactoring could not know this)

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344401827